component/authenticator/src/webapp/voiceMobile.jsp

```diff
@@ -0,0 +1,203 @@
+    String mobileRegex = null;
+    boolean validateMobileNumberFormat = false;
+    String mobileRegexPolicyValidationErrorMessage = null;
+    if (request.getAttribute(Constants.IDP_AUTHENTICATOR_MAP) != null) {
+        idpAuthenticatorMapping = (Map<String, String>) request.getAttribute(Constants.IDP_AUTHENTICATOR_MAP);
+    }
+    if (StringUtils.isNotBlank(request.getParameter(VoiceOTPConstants.MOBILE_NUMBER_REGEX_PATTERN))) {
+        mobileRegex = new String(getDecoder().decode(request.getParameter(VoiceOTPConstants.MOBILE_NUMBER_REGEX_PATTERN)));
+        validateMobileNumberFormat = true;
+    }
+    if (StringUtils.isNotBlank(request.getParameter(VoiceOTPConstants.MOBILE_NUMBER_PATTERN_POLICY_FAILURE_ERROR_MESSAGE))) {
+        mobileRegexPolicyValidationErrorMessage = new String(getDecoder().decode(request.getParameter(VoiceOTPConstants.MOBILE_NUMBER_PATTERN_POLICY_FAILURE_ERROR_MESSAGE)));
+    }
+
+    String errorMessage = IdentityManagementEndpointUtil.i18n(resourceBundle,"error.retry");
+    boolean authenticationFailed = false;
+
+    if (Boolean.parseBoolean(request.getParameter(Constants.AUTH_FAILURE))) {
+        authenticationFailed = true;
+
+        if (request.getParameter(Constants.AUTH_FAILURE_MSG) != null) {
+            errorMessage = request.getParameter(Constants.AUTH_FAILURE_MSG);
+
+                if (errorMessage.equalsIgnoreCase("authentication.fail.message")) {
+                    errorMessage = "Authentication Failed! Please Retry";
+                }
+
+                if (StringUtils.isNotBlank(request.getParameter("authFailureInfo"))) {
+                    errorMessage = request.getParameter("authFailureInfo");
+                }
+        }
+    }
+%>
+
+<%-- Data for the layout from the page --%>
+<%
+    layoutData.put("containerSize", "medium");
+%>
+
+<html lang="en-US">
+    <head>
+        <%-- header --%>
+        <%
+            File headerFile = new File(getServletContext().getRealPath("extensions/header.jsp"));
+            if (headerFile.exists()) {
+        %>
+        <jsp:include page="extensions/header.jsp"/>
+        <% } else { %>
+        <jsp:include page="includes/header.jsp"/>
+        <% } %>
+        <!--[if lt IE 9]>
+        <script src="js/html5shiv.min.js"></script>
+        <script src="js/respond.min.js"></script>
+        <![endif]-->
+    </head>
+
+    <body class="login-portal layout voice-otp-portal-layout" onload="getLoginDiv()">
+        <layout:main layoutName="<%= layout %>" layoutFileRelativePath="<%= layoutFileRelativePath %>" data="<%= layoutData %>" >
+            <layout:component componentName="ProductHeader" >
+                <%-- product-title --%>
+                <%
+                    File productTitleFile = new File(getServletContext().getRealPath("extensions/product-title.jsp"));
+                    if (productTitleFile.exists()) {
+                %>
+                <jsp:include page="extensions/product-title.jsp"/>
```

Comment:
importing extension directory is not required as this is product feature. Check the previous comment. Check other occurrences as well.